WordPress/gutenberg
# Reveal block boundaries on hover in the Site Editor

This PR roughly implements parts of a [prototype](https://cdpn.io/jameskoster/debug/ZEOgLxV) I worked on for https://github.com/WordPress/gutenberg/issues/27239, to see how it feels more holistically.

![boundaries](https://user-images.githubusercontent.com/846565/100227608-4bf57f00-2f19-11eb-81dd-5e4f817223f6.gif)

Block boundaries are exposed on hover making it easier to understand what will be selected upon click. This is important in the Site Editor particularly, because block nesting features prominently in that context.

A different color is applied to template parts, to indicate that they are different to regular blocks.

This feels like a net positive change to me, although I would say that the `is-hovered` class seems to be somewhat brittle. Moving the cursor around quickly can trip it up. It doesn't occur all that often, but if there's any way to improve performance there it would be worth exploring.

To test:

* Load up a template in the Site Editor
* Apply hover, select, focus, select+focus states to various blocks and ensure the following:

1. Un-selected, hovered blocks display a 1px blue outline
2. Selected + focussed blocks (IE on-click) display a 1.5px blue outline
3. Selected un-focussed blocks display a 1px black outline
